applications/callflow/src/module/cf_dynamic_cid.erl

```diff
@@ -264,3 +263,19 @@ get_list_entry(Data, Call) ->
             lager:info("failed to load match list box ~s: ~p", [ListId, _Reason]),
             E
     end.
+
+-spec get_lists_entry(wh_json:object(), whapps_call:call()) -> proplist() | {'error', couch_mgr:couchbeam_error()}.
+get_lists_entry(Data, Call) ->
+    ListId = wh_json:get_ne_value(<<"id">>, Data),
+    AccountDb = whapps_call:account_db(Call),
+    case couch_mgr:get_results(AccountDb,<<"lists/entries">>,[{'key', ListId}]) of
+        {'ok', Entries} ->
+	    CaptureGroup = whapps_call:kvs_fetch('cf_capture_group', Call),
+	    <<CIDKey:2/binary, Dest/binary>> = CaptureGroup,
+            [{NewCallerIdName, NewCallerIdNumber}|_] = [{wh_json:get_binary_value([<<"value">>, <<"cid_name">>], Entry), wh_json:get_value([<<"value">>, <<"cid_number">>], Entry)}
+                                                        || Entry <- Entries, wh_json:get_binary_value([<<"value">>, <<"cid_key">>], Entry) == CIDKey],
```

Comment:
If the order doesn't matter, consider using a foldl instead (saves some computations).
Otherwise, make sure that taking the first item won't bite you.